src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -506,10 +506,13 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             // Don't attempt preferred kernel search for interactive window, but do make sure we
             // load all our controllers for interactive window
             const notebookMetadata = getNotebookMetadata(document);
-            if (document.notebookType === JupyterNotebookView) {
-                const resourceType = getResourceType(document.uri);
-                const isPythonNbOrInteractiveWindow =
-                    isPythonNotebook(notebookMetadata) || resourceType === 'interactive';
+            const resourceType = getResourceType(document.uri);
+            const isPythonNbOrInteractiveWindow = isPythonNotebook(notebookMetadata) || resourceType === 'interactive';
+            if (document.notebookType === JupyterNotebookView && !this.isLocalLaunch && isPythonNbOrInteractiveWindow) {
```

Comment:
When dealing with remotes, always give preference to remote kernels, 
This is how it used to work, 

Basically reverting to previous behavior to enur tests pass successfully.
Changes to this can be made here along with the fixes to the tests https://github.com/microsoft/vscode-jupyter/issues/9834